src/coreclr/vm/gcenv.ee.common.cpp

```diff
@@ -149,6 +154,22 @@ void GcEnumObject(LPVOID pData, OBJECTREF *pObj, uint32_t flags)
     Object ** ppObj = (Object **)pObj;
     GCCONTEXT   * pCtx  = (GCCONTEXT *) pData;
 
+    if (g_pConfig->GetEnableGCHoleMonitoring() && ((flags & GC_CALL_PINNED) == 0) && pCtx->sc->promotion)
+    {
+        if (doubleReportTrackingIndex == doubleReportTrackingSize)
+        {
+            doubleReportTrackingSize = (doubleReportTrackingSize == 0) ? DoubleReportTrackingInitialSize : doubleReportTrackingSize * 2;
+            doubleReportTracking = (void**)realloc(doubleReportTracking, doubleReportTrackingSize * sizeof(void*));
+        }
+        for (int i = 0; i < doubleReportTrackingIndex; i++)
```

Comment:
Array has quadratic complexity. Would hashtable be better?

Reply:
I didn't think it is worth complicating this stuff. My intention was to have this as a separate tool that would be only turned on when a runtime developer would investigate a GC hole. I would expect this array to be small in vast majority of the cases, so linear scan seemed to be good enough.

Reply:
I have changed it to use a hash table